refactor!: remove redundant sending of translations

barredterra opened the pull request:
This code was used to load the specific translations needed for a particular view. A while ago, we've started sending **all** translations to the frontend, regardless of the view (https://github.com/frappe/frappe/pull/18764). Therefore, this became redundant.

The only exception are web forms, which don't receive all translations. I've added a dedicated mechanism to load the required translations for web form.


- [x] update https://github.com/frappe/frappe/wiki/Migrating-to-nightly-version
	- removed functionality of `get_translated_dict` hook
	- stopped translating currency codes and timezones
		- deprecated `get_translated_dict` method. What's remaining is now called `get_translated_countries`
	- general availability of translations for country names

barredterra committed:
chore!: remove redundant translation

ankush commented:
I left all of this because there were still places where it was used... ig I'll just recheck where it's used and remove this finally.

ankush commented on frappe/translate.py:
```diff
@@ -159,63 +159,6 @@ def get_lang_dict():
-
-		messages = deduplicate_messages(messages)
-		message_dict = make_dict_from_messages(messages, load_user_translation=False)
-		message_dict.update(get_dict_from_hooks(fortype, name))
```
This hook seems to be used still. Should we preserve this somehow (or merge it in default messages (?))

https://github.com/frappe/frappe/blob/daca4dd31c42ac203cca666d1d4239d6d1de4db3/frappe/geo/country_info.py#L41-L64 


https://github.com/frappe/frappe/blob/daca4dd31c42ac203cca666d1d4239d6d1de4db3/frappe/hooks.py#L272-L275

ankush replied:
Everything else LGTM.

barredterra replied:
Thanks, I'll check this

barredterra replied:
The `get_translated_dict` hook [only ever pointed](https://sourcegraph.com/search?q=context%3Aglobal+file%3Ahooks.py+get_translated_dict&patternType=standard&sm=1&groupBy=repo) to `frappe.geo.country_info.get_translated_dict`, which returns  translations for timezones, countries and currencies (provided by babel). It was not included in hooks boilerplate, so barely used by other apps.

I removed the hook and included these translations in `get_all_translations`. This also reduces confusion ("Why do countries get translated in the Setup Wizard but not in my custom doctype?", "Why are timezones displayed differently in Setup Wizard and User?", etc).

barredterra replied:
But I'm not sure if we need translations of currency codes and timezones at all. If in doubt, I'd prefer to ditch them.

ankush replied:
I don't think we need them too. 😅 

Change in money_in_words might not be acceptable too. Currency symbol is perfectly fine instead of full currency name there.

barredterra replied:
I removed the translations of currency codes and timezones. Translations for country names remain available everywhere.
The output of `money_in_words` should stay unchanged now - I reverted my changes to the respective test.

barredterra committed:
refactor!: remove "get_translated_dict" hook

Instead, include the results from
`frappe.geo.country_info.get_translated_dict` in `get_all_translations`.

barredterra committed:
test: money_in_words uses translated currency

... and now we have translations.

barredterra committed:
Revert "test: money_in_words uses translated currency"

This reverts commit e07d594ac1ce4b120c0fceb5d23377cb981fe820.

barredterra committed:
refactor!: stop translating timezone and currency

barredterra committed:
refactor: rename `get_translated_dict`

... to `get_translated_countries`

ankush commented:
@barredterra can you update migration guide please.
